Lib/_collections_abc.py

```diff
@@ -481,15 +481,8 @@ def __getitem__(self, item):
         # rather than the default types.GenericAlias object.  Most of the
         # code is copied from typing's _GenericAlias and the builtin
         # types.GenericAlias.
-
         if not isinstance(item, tuple):
             item = (item,)
-        # A special case in PEP 612 where if X = Callable[P, int],
-        # then X[int, str] == X[[int, str]].
-        if (len(self.__parameters__) == 1
-                and _is_param_expr(self.__parameters__[0])
-                and item and not _is_param_expr(item[0])):
-            item = (item,)
```

Comment:
This makes me slightly nervous, but I can't find any changes in behaviour from removing this code.

@Fidget-Spinner and @serhiy-storchaka, can either of you think of anything bad that could happen if we remove this code? The whole of `test_typing` passes with it removed.

Reply:
Yes, me too. Let's wait for others to comment.

Reply:
It makes me nervous too. In case you're looking for an example that triggers this path, here's one:
```py
from typing import ParamSpec
P = ParamSpec("P")
from collections.abc import Callable
C = Callable[[P], int]
print(C[str])
```
However I agree that if I just comment this out, the output is the same. It's possible that @serhiy-storchaka fixed the issue in the C code (the `super()` method) -- the code here is from Aug 4th, 2021, while he touched `_Py_subs_parameters` on May 8th, 2022.

Reply:
Yes, I found examples that triggered this code path, but for all of the examples I found, I couldn't find any differences in behaviour with this code deleted (repr, `__args__`, `__parameters__`, equality)

Reply:
Someone could trace through the C code that gets called and see if it does the right thing.

Reply:
Looks like `_unpack_args` C function which was added in May now handles this case: https://github.com/python/cpython/blame/2dc94634b50f0e5e207787e5ac1d56c68b22c3ae/Objects/genericaliasobject.c#L361

It unpacks nested tuples to a flat format, so - this code is not needed anymore.

Reply:
I don't suppose the pure-Python code could be useful for PyPy, could it? (Don't see how it could, just throwing that out there)